pkg/ccl/backupccl/backup_job.go

```diff
@@ -988,6 +988,39 @@ func maybeRelocateJobExecution(
 	return nil
 }
 
+// checkForExistingManifestDifferentClusterID returns an error if it finds a backup
+// manifest checkpoint file for this backup job, created by a different
+// cluster.
+//
+// TODO(msbutler): deduplicate this method with b.readManifestOnResume(). This method
+// is kept separate to minimize the backport impact to address
+// https://github.com/cockroachdb/cockroach/issues/112114
+func checkForExistingManifestWithDifferentClusterID(
+	ctx context.Context,
+	execCfg *sql.ExecutorConfig,
+	mem *mon.BoundAccount,
+	details jobspb.BackupDetails,
+	kmsEnv cloud.KMSEnv,
+	user username.SQLUsername,
+) error {
+	defaultConf, err := cloud.ExternalStorageConfFromURI(details.URI, user)
```

Comment:
Can we just do `if c := job.Payload().CreationClusterID; c != uuid.Nil && c != jobExec.ExtendedEvalContext().ClusterID { return errors.Newf(...)` rather than introducing more external IO?